### PR TITLE
Benchmarks

### PR DESCRIPTION
**Pull request by saikrishna-1996**
WIP

### CODE REVIEW

File: actors/dqn.py [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+
+        environment_specs = EnvironmentSpecs.deserialize(config.environment_specs)
+        observation_space = environment_specs.get_observation_space()
+        action_space = environment_specs.get_action_space(seed=rng.integers(9999))
+
+        assert isinstance(action_space.gym_space, Discrete)
+
+        model, _, _ = await actor_session.model_registry.retrieve_version(
+            DQNModel, config.model_id, config.model_version
+        )
```
Comment (wduguay-air) [inline actors/dqn.py:258]: The word version was completely disassociated from the `model_registry_v2`. All instances of `version` or `model_version` are now `iteration` or `model_iteration`. This is the case for `config.model_iteration`.

File: actors/dqn.py [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+                    actor_session.do_action(action_space.serialize(action_space.create()))
+                    continue
+
+                if (
+                    config.model_update_frequency > 0
+                    and self.samples_since_update > 0
+                    and self.samples_since_update % config.model_update_frequency == 0
+                ):
+                    model, _, _ = await actor_session.model_registry.retrieve_version(
+                        DQNModel, config.model_id, config.model_version
```
Comment (wduguay-air) [inline actors/dqn.py:275]:
```suggestion
                        DQNModel, config.model_id, config.model_iteration
```

File: config/experiment/dqn/cartpole.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  - override /services/actor:
+      - dqn
+  - override /services/environment: cartpole
+
+run:
+  class_name: actors.dqn.DQNTraining
+  seed: 618
+
+  # Archiving
+  archive_model: True
```
Comment (wduguay-air) [inline config/experiment/dqn/cartpole.yaml:12]: The `archive_model` and `archive_frequency` parameters are currently not used in the training class.

File: actors/dqn.py [hunk 4]
```diff
@@ -0,0 +1,524 @@
...
+    @staticmethod
+    def serialize_model(model) -> bytes:
+        stream = io.BytesIO()
+        torch.save(
+            (
+                model.network.state_dict(),
+                model.epsilon,
+                model.get_model_user_data(),
+            ),
+            stream,
```
[the remark this answers is not in the document]
Comment (saikrishna-1996) [reply actors/dqn.py:211]: No, `self.network` contains the full network

File: actors/dqn.py [hunk 5]
```diff
@@ -0,0 +1,524 @@
...
+        environment_specs = EnvironmentSpecs.deserialize(config.environment_specs)
+        observation_space = environment_specs.get_observation_space()
+        action_space = environment_specs.get_action_space(seed=rng.integers(9999))
+
+        assert isinstance(action_space.gym_space, Discrete)
+
+        model, _, _ = await actor_session.model_registry.retrieve_version(
+            DQNModel, config.model_id, config.model_version
+        )
+        model.network.eval()
```
[the remark this answers is not in the document]
Comment (saikrishna-1996) [reply actors/dqn.py:259]: No, `model.network` is the full network

File: actors/dqn.py [hunk 6]
```diff
@@ -0,0 +1,524 @@
...
+        elif len(x.shape) == 5:
+            x = x.reshape(x.size(0), -1, x.size(-2), x.size(-1))
+        x = x.float()
+        x = x / self._normalization_factor
+        x = self.conv(x)
+        x = self.mlp(x)
+        return x
+
+
+class DQNNetwork(nn.Module):
```
[the remark this answers is not in the document]
Comment (saikrishna-1996) [reply actors/dqn.py:145]: This allows for the base_network to be either fully connected (MLPNetwork) or convolutional (ConvNetwork) (with any arbitrary sizes passed in as params) and adds the final output layer

File: config/services/actor/dqn.yaml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+simple_dqn:
```
Comment (wduguay-air) [inline config/services/actor/dqn.yaml:1]:
```suggestion
dqn:
```

File: actors/dqn.py [hunk 8]
```diff
@@ -0,0 +1,534 @@
...
+import torch
+from torch import nn
+from gym.spaces import Discrete, utils
+
+from cogment_verse.specs import AgentConfig, cog_settings, EnvironmentConfig, EnvironmentSpecs
+
+from cogment_verse.constants import PLAYER_ACTOR_CLASS
+
+from cogment_verse import Model, TorchReplayBuffer  # pylint: disable=abstract-class-instantiated
+
```
Comment (wduguay-air) [inline actors/dqn.py:36]: Seed stuff should be in configs.

File: actors/dqn.py [hunk 9]
```diff
@@ -0,0 +1,534 @@
...
+    def __init__(
+        self,
+        model_id,
+        environment_implementation,
+        num_input,
+        num_output,
+        num_hidden_nodes,
+        epsilon,
+        dtype=torch.float,
+        version_number=0,
```
Comment (wduguay-air) [inline actors/dqn.py:175]:
```suggestion
        iteration=0,
```

File: actors/dqn.py [hunk 10]
```diff
@@ -0,0 +1,534 @@
...
+    ):
+        super().__init__(model_id, version_number)
```
Comment (wduguay-air) [inline actors/dqn.py:177]:
```suggestion
        super().__init__(model_id, iteration)
```

File: actors/dqn.py [hunk 11]
```diff
@@ -0,0 +1,534 @@
...
+                            cogment.ActorParameters(
+                                cog_settings,
+                                name="player",
+                                class_name=PLAYER_ACTOR_CLASS,
+                                implementation="actors.dqn.DQNActor",
+                                config=AgentConfig(
+                                    run_id=run_session.run_id,
+                                    seed=self._cfg.seed + trial_idx,
+                                    model_id=model_id,
+                                    model_version=-1,
```
Comment (wduguay-air) [inline actors/dqn.py:450]:
```suggestion
                                    model_iteration=-1,
```
